extensions/amp-access-scroll/0.1/scroll-relay.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getData} from '../../../src/event-helper';
+import {listen} from '../../../src/3p-frame-messaging';
+
+/**
+ * Provides cross-frame postMessage for scroll frames.
+ */
+export class Relay {
+  /**
+   * @param {string} domain - origin restriction for postMessage
+   */
+  constructor(domain) {
+    /** @private @type {!Array<Window>} */
+    this.frames_ = [];
+    /** @private @type {!Array<function(JsonObject)>} */
+    this.listeners_ = [];
+    /** @private */
+    this.origin_ = domain;
+
+    this.onMessage_ = this.onMessage_.bind(this);
+
+    listen(window, 'message', this.onMessage_);
+  }
+
+  /**
+   * @param {!Event} e
+   * @private
+   */
+  onMessage_(e) {
+    const data = /** @type {!JsonObject} */ (getData(e));
+    const fromScrollOrigin = e.origin === this.origin_;
+    const isScrollAmpMessage = '_scramp' in data;
```

Comment:
If `data == null`, this will fail. Let's do `typeof data == 'object' && ...`

Reply:
Updated